[vamp-sdk] update to 2.10; switch download URL to GitHub; fix path of installed headers

Be-ing opened the pull request:
This works around a bad SSL certificate on
https://code.soundsoftware.ac.uk :

Error: Failed to download from mirror set:
https://code.soundsoftware.ac.uk/attachments/download/2589/vamp-plugin-sdk-2.9.0.zip: % Total    % Received % Xferd  Average Speed   Time    Time Time  Current
                                 Dload  Upload   Total   Spent Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

**Describe the pull request**

- #### What does your PR fix?  
  Fixed outdated port and works around bad SSL certificate

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

Be-ing committed:
[vamp-sdk] update to 2.10; switch download URL to GitHub

This works around a bad SSL certificate on
https://code.soundsoftware.ac.uk :

Error: Failed to download from mirror set:
https://code.soundsoftware.ac.uk/attachments/download/2589/vamp-plugin-sdk-2.9.0.zip: % Total    % Received % Xferd  Average Speed   Time    Time Time  Current
                                 Dload  Upload   Total   Spent Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

Be-ing commented:
x64-windows-static failure seems unrelated? Azure Pipelines says there is a regression in colmap:x64-windows-static...

dg0yt commented:
> x64-windows-static failure seems unrelated? Azure Pipelines says there is a regression in colmap:x64-windows-static...

~~~
ninja: error: 'D:/installed/x64-windows-static/debug/lib/openblas.lib', needed by 'src/exe/colmap.exe', missing and no known rule to make it
~~~
This is unrelated. It happened to another PR already: #19551.

ras0219 commented:
This GitHub repo (c4dm/vamp-plugin-sdk) looks official enough to me to depend upon

Be-ing commented:
I trust GitHub to keep their SSL certificate valid more than the website for this package.

Be-ing committed:
[vamp-sdk] fix path of installed headers

These are the paths installed by the vamp-plugin-sdk-devel Fedora
package:

/usr/include/vamp
/usr/include/vamp-hostsdk
/usr/include/vamp-hostsdk/Plugin.h
/usr/include/vamp-hostsdk/PluginBase.h
/usr/include/vamp-hostsdk/PluginBufferingAdapter.h
/usr/include/vamp-hostsdk/PluginChannelAdapter.h
/usr/include/vamp-hostsdk/PluginHostAdapter.h
/usr/include/vamp-hostsdk/PluginInputDomainAdapter.h
/usr/include/vamp-hostsdk/PluginLoader.h
/usr/include/vamp-hostsdk/PluginSummarisingAdapter.h
/usr/include/vamp-hostsdk/PluginWrapper.h
/usr/include/vamp-hostsdk/RealTime.h
/usr/include/vamp-hostsdk/host-c.h
/usr/include/vamp-hostsdk/hostguard.h
/usr/include/vamp-hostsdk/vamp-hostsdk.h
/usr/include/vamp-sdk
/usr/include/vamp-sdk/FFT.h
/usr/include/vamp-sdk/Plugin.h
/usr/include/vamp-sdk/PluginAdapter.h
/usr/include/vamp-sdk/PluginBase.h
/usr/include/vamp-sdk/RealTime.h
/usr/include/vamp-sdk/plugguard.h
/usr/include/vamp-sdk/vamp-sdk.h
/usr/include/vamp/vamp.h

PhoebeHui reviewed (approved):
@Be-ing, thanks for your PR!

cannam commented:
It's after the fact I know, but I just want to agree that this is the right thing to do. There have been ongoing problems with the previous server, and the Github repo is equally "official".
